### PR TITLE
Have `safe_get` raise HTTP exception for retrying to work

### PR DESCRIPTION
**Pull request by duynguyen158**
## Description

The 404 errors recently encountered (see #155) prompt a need to retry requesting for at least a few times and with exponential backoff. 

Since the `safe_get` function in `sites/helpers.py` already has the ability to retry with exponential backoff via its `retry` decorator, this PR have its `page` object raise an exception if a 4xx or 5xx error occurs, thus allowing the decorator to actually work. This is basically what the `validate_response_code` function in `sites/helpers.py` does, so this function (as well as the site-specific code that references it) is now deleted.

## Testing

Two unit tests for `validate_response_code` are deleted. All the other ones passed.

### CODE REVIEW

File: sites/helpers.py [hunk 1]
```diff
@@ -84,8 +82,19 @@ def safe_get(
...
     page = req.get(url, timeout=TIMEOUT_SECONDS, params=params, headers=default_headers)
+
+    # Many times, the request hits a 4xx or 5xx, but no exception is raised
+    # This makes sure an exception is raised and allows the retry decorator to work.
+    #
+    # Some notes:
+    # - Since we already have exponential backoff, no need to treat 429 error any differently.
+    # - During the last retry attempt, if this still raises an exception, that exception's already handled
+    # by a try-except block in each site's fetch_article method.
+    page.raise_for_status()
```
Comment (raaidarshad) [inline sites/helpers.py:93]: ❓ Do we know how this is handled if an exception is raised? I see `fetch_article` in each partner site has it wrapped in a `try/except`, but there are uses in `bulk_fetch` that aren't wrapped AFAICT.

**Review by raaidarshad — CHANGES_REQUESTED** [review]
One main question, otherwise I think it looks good; touches enough things via use that its a bit hard to tell fully confidently. Resisting more code clean up hehe

File: sites/helpers.py (hunk 1 above)
Comment (duynguyen158) [reply sites/helpers.py:93]: Thanks for pointing that out; didn't realize that and just assumed `bulk_fetch` would just use `fetch_article`.

Comment (duynguyen158) [reply sites/helpers.py:93]: Wrapping these now.

**Review by raaidarshad — APPROVED** [review]
lgtm

**Pull request merged**